model/swagger.yml

```diff
@@ -1149,8 +1149,6 @@ components:
         syntax: limit=10
 
         behavior: maximum number of matching results returned, for pagination
-        
-        note: limit=0 returns just the summary
```

Comment:
I'll keep that note, although data is still returned with length 0, I would like that we made obvious to the users that limit=0 is the way to get the summary only.

Reply:
Is that useful information to provide to the user?  "If you request zero data you'll get zero data" seems pointless, and specifying that you'll get just the summary (which is actually the search metadata) seems to indicate that something nonstandard occurs.  Your call, though.

Reply:
@tloubrieu-jpl ☝️

Reply:
@alexdunnjpl we wanted a way for the user to get the metadata for a set of results, beyond the pagination limits, without requesting them all. That is useful for a user who would like to know the set of properties which can be requested in this products.

The use case is:
1. user want to explore the product's of a collection (e.g. 100000s products)
2. the user needs to know which properties are used to describe these products
3. the user can apply search criteria on some of the available properties to subset the collection of products (these properties cannot be guessed by the user)

That is for step 2 that we initially defined the `summaryOnly` keyword, and then we decided to remove this keyword to use limit=0 instead. That being said it sounds like the summary does not provide the description for the full request results beyond pagination limits but only for the current page.

Reply:
@tloubrieu-jpl if I understand you correctly, it sounds like `limit=0` _is_ supposed to behave in a different/non-standard manner, then? Specifically, to provide the union of the set of properties present in each product for the given query? (That's a non-homogeneous collection of property sets, yeah? Otherwise the user could just get it with `limit=1` and extract the property set themselves)

It currently doesn't, iirc, so sounds like there's a need to make some kind of fix if that's necessary, but that almost seems like a job for a subroute like `products/properties?{someQuery}` rather than serving it in the case of `products?{someQuery}&limit=0`?

Reply:
@alexdunnjpl I was unclear, I am not looking for a non standard management of limit=0. I believe what is rather needed to be updated is the way the summary is prepared. For the use case I described above, the summary need to be made from the products matching the query (q, keyword) but ignoring the pagination parameters (start, limit). I will make a more detailed ticket for that, because all the summary properties are not equivalent.